src/drivers/cmakeDriver.ts

```diff
@@ -1742,12 +1820,33 @@ export abstract class CMakeDriver implements vscode.Disposable {
                 }
             }
             const useBuildTask: boolean = this.config.buildTask && isBuildCommand === true;
-            const exeOpt: proc.ExecutionOptions = { environment: buildcmd.build_env, outputEncoding: outputEnc, useBuildTask: useBuildTask };
-            const child = this.executeCommand(buildcmd.command, buildcmd.args, consumer, exeOpt);
-            this._currentBuildProcess = child;
-            await child.result;
-            this._currentBuildProcess = null;
-            return child;
+            if (useBuildTask) {
+                const task = await this._findBuildTask(targets);
+                if (task) {
+                    this._requestedTargets = targets;
+                    await vscode.tasks.executeTask(task);
```

Comment:
The other path has a way to cancel the build via the UI and block other operations (when `this._currentBuildProcess` is set).  Is there a way to get a handle to the task so that this path can do something similar?  I don't believe the current implementation works correctly, but it would be good to fix it while we're updating the code if possible.

Reply:
Current implementation does not work correctly. Even more - it does not wait for task processing to be finished (it startą asynchronous proces if task execution and exits). In this PR I have change it - currently extension will return Promise with task execution and will wait.
I will take a closer look for cancellation - maybe it works out of the box now :)

Reply:
In last commit I've added possibility to cancel.